staker/staker.go

```diff
@@ -257,9 +260,43 @@ func (s *Staker) Initialize(ctx context.Context) error {
 			return err
 		}
 
-		return s.blockValidator.AssumeValid(stakedInfo.AfterState().GlobalState)
+		return s.blockValidator.InitAssumeValid(stakedInfo.AfterState().GlobalState)
+	}
+	return nil
+}
+
+func (s *Staker) checkLatestStaked(ctx context.Context) error {
+	latestStaked, _, err := s.validatorUtils.LatestStaked(&s.baseCallOpts, s.rollupAddress, s.wallet.AddressOrZero())
+	if err != nil {
+		return fmt.Errorf("couldn't get LatestStaked: %w", err)
+	}
+	stakerLatestStakedNodeGauge.Update(int64(latestStaked))
+	if latestStaked == 0 {
+		return nil
+	}
+
+	stakedInfo, err := s.rollup.LookupNode(ctx, latestStaked)
+	if err != nil {
+		return fmt.Errorf("couldn't look up latest node: %w", err)
 	}
 
+	stakedGlobalState := stakedInfo.AfterState().GlobalState
+	caughtUp, count, err := GlobalStateToMsgCount(s.inboxTracker, s.txStreamer, stakedGlobalState)
+	if err != nil {
+		if errors.Is(err, ErrGlobalStateNotInChain) && s.fatalErr != nil {
+			fatal := fmt.Errorf("latest staked not in chain: %w", err)
+			s.fatalErr <- fatal
```

Comment:
I think this fatal error might prevent the node from performing a reorg it needs to do to correct its invalid state (because it shuts down before it's able to reorg). Perhaps this should only be fatal if the stakedInfo's inbox accumulator matches the inbox tracker's record, otherwise just return an error here.

Reply:
Let's about this.
We've already read the batch for this data and it's quite old (feed messages won't get us "caught up")
Most chances are there was some weird one-off error in calculating state that we can't really recover from.
I think being very noisy with this error is worth having to go through manual hoops in case it is recoverable.

Reply:
Makes sense. I guess if someone is error-looping on this they could always just disable the staker for a bit until the reorg happens.